### PR TITLE
More robust stop-timer tests implementation

### PR DESCRIPTION
**Pull request by Peque**
These tests were failing from time to time. Now they are more robust and fast (avoiding unnecessary time sleeps).

### CODE REVIEW

**Comment by Peque** [comment]
@ocaballeror Travis fails due to the linger test failing... :joy: (waiting for #260 to fix that). :stuck_out_tongue_winking_eye:

**Comment by Peque** [comment]
@ocaballeror AppVeyor keeps failing on `ns.shutdown()` on test teardown. I try to update #259 with new occurrences. I say we fix all these CI issues before adding new features in this 0.7.0 cycle.

**Comment by ocaballeror** [comment]
Yes! I never liked the arbitrary `time.sleep()`.

Let me test it a few times in Travis to make sure it works, you have my approval for everything else.

**Comment by ocaballeror** [comment]
This test looks like it could benefit from this as well: https://ci.appveyor.com/project/ocaballeror/osbrain/build/1.0.42/job/xbsqth4u9m1597s4#L761

Should we change it and include it in this PR?

File: osbrain/tests/test_helper.py [hunk 1]
```diff
@@ -118,25 +118,44 @@ def test_logger_received_count(agent_logger):
...
+    (dict(attribute={'foo'}, data='foo'), True),
+    (dict(attribute={'foo'}, value={'foo'}), True),
+    (dict(attribute=42, value=14), False),
+    (dict(attribute=42, value=42), True),
+    (dict(attribute=[1, 2, 3], endswith=[2, 3]), True),
+    (dict(attribute=[1, 2, 3], endswith=[3, 3]), False),
+    (dict(attribute='abcde', endswith='cde'), True),
+    (dict(attribute='abcde', endswith='ade'), False),
+], ids=[
+    'List length missmatch',
```
Comment (ocaballeror) [inline osbrain/tests/test_helper.py]: 'Mismatch'

Comment (ocaballeror) [inline osbrain/tests/test_helper.py]: 'mismatch'

File: osbrain/tests/test_helper.py [hunk 2]
```diff
@@ -118,25 +118,44 @@ def test_logger_received_count(agent_logger):
...
+    'Data not in list',
+    'List exact value',
+    'Data in list and list length',
+    'Data not in list with length match',
+    'Data in list with length missmatch',
```
Comment (ocaballeror) [inline osbrain/tests/test_helper.py]: 'mismatch'

File: osbrain/tests/test_helper.py [hunk 3]
```diff
@@ -118,25 +118,44 @@ def test_logger_received_count(agent_logger):
...
+    'Dictionary attribute length',
+    'Key in dictionary',
+    'Dictionary exact value',
+    'Integer value missmatch',
```
Comment (ocaballeror) [inline osbrain/tests/test_helper.py]: 'mismatch'

**Comment by Peque** [comment]
Good catch with the typos! ^^

About the error in `osbrain/tests/test_agent_async_requests.py`, I had a look at it and I think it should be a separate issue. It is a good idea to change that, but if we do I'd rather re-think the whole `osbrain/tests/test_agent_async_requests.py` file, not just that particular test. So I say we merge this as-is if there are no other objections.

**Pull request merged**